src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -2509,5 +2512,616 @@ uint8_t RadioSpinel<InterfaceType, ProcessContextType>::GetCslUncertainty(void)
+        start += snprintf(start, static_cast<size_t>(end - start), ", rxRequest:%u", metrics.mNumRxRequest);
+        start +=
+            snprintf(start, static_cast<size_t>(end - start), ", rxGrantImmediate:%u", metrics.mNumRxGrantImmediate);
+        start += snprintf(start, static_cast<size_t>(end - start), ", rxGrantWait:%u", metrics.mNumRxGrantWait);
+        start += snprintf(start, static_cast<size_t>(end - start), ", rxGrantWaitActivated:%u",
+                          metrics.mNumRxGrantWaitActivated);
+        start += snprintf(start, static_cast<size_t>(end - start), ", rxGrantWaitTimeout:%u",
+                          metrics.mNumRxGrantWaitTimeout);
+        start += snprintf(start, static_cast<size_t>(end - start), ", rxGrantDeactivatedDuringRequest:%u",
+                          metrics.mNumRxGrantDeactivatedDuringRequest);
+        start += snprintf(start, static_cast<size_t>(end - start), ", rxDelayedGrant:%u", metrics.mNumRxDelayedGrant);
+        start += snprintf(start, static_cast<size_t>(end - start), ", avgRxRequestToGrantTime:%u",
+                          metrics.mAvgRxRequestToGrantTime);
+        start += snprintf(start, static_cast<size_t>(end - start), ", rxGrantNone:%u", metrics.mNumRxGrantNone);
+        start += snprintf(start, static_cast<size_t>(end - start), ", stopped:%u", metrics.mStopped);
+        start += snprintf(start, static_cast<size_t>(end - start), ", grantGlitch:%u", metrics.mNumGrantGlitch);
+    }
+    break;
+
+    case SPINEL_PROP_MAC_SCAN_MASK:
+    {
+        uint8_t       channels[16];
+        spinel_size_t size;
+
+        unpacked = spinel_datatype_unpack(data, len, SPINEL_DATATYPE_DATA_S, channels, &size);
+        VerifyOrExit(unpacked > 0, error = OT_ERROR_PARSE);
+        start += snprintf(start, static_cast<size_t>(end - start), ", channels:");
+
+        for (uint8_t i = 0; i < size; i++)
+        {
+            start += snprintf(start, static_cast<size_t>(end - start), "%u ", channels[i]);
+        }
+    }
+    break;
+
+    case SPINEL_PROP_RCP_ENH_ACK_PROBING:
+    {
+        uint16_t saddr;
+        uint8_t  m8[OT_EXT_ADDRESS_SIZE];
+        uint8_t  flags;
+
+        unpacked = spinel_datatype_unpack(
+            data, len, SPINEL_DATATYPE_UINT16_S SPINEL_DATATYPE_EUI64_S SPINEL_DATATYPE_UINT8_S, &saddr, m8, &flags);
+
+        VerifyOrExit(unpacked > 0, error = OT_ERROR_PARSE);
+        start += snprintf(start, static_cast<size_t>(end - start),
+                          ", saddr:%04x, extaddr:%02x%02x%02x%02x%02x%02x%02x%02x, flags:0x%02x", saddr, m8[0], m8[1],
+                          m8[2], m8[3], m8[4], m8[5], m8[6], m8[7], flags);
+    }
+    break;
+    }
+
+exit:
+    return error;
+}
+
+template <typename InterfaceType, typename ProcessContextType>
+void RadioSpinel<InterfaceType, ProcessContextType>::LogSpinelFrame(const uint8_t *aFrame, uint16_t aLength, bool aTx)
+{
+    constexpr uint16_t kLogBufferSize      = 512;
```

Comment:
Should we try to break the frame log info into multiple lines? (avoid truncation of log line)
- Our default `OPENTHREAD_CONFIG_LOG_MAX_SIZE` is 150 chars.
- I also recall, syslog in some embedded platform is ~450 chars per line

It may also look nicer if we log info in multiple lines (e.g. for coex counters)?

Reply:
I have updated the code to break the log message into multiple lines.